lib/middleware.js

```diff
@@ -86,11 +87,25 @@ module.exports = function(watcher) {
       response.setHeader('Content-Length', stat.size)
       response.setHeader('Content-Type', type)
 
-      // read file sync so we don't hold open the file creating a race with
-      // the builder (Windows does not allow us to delete while the file is open).
-      buffer = fs.readFileSync(filename)
-      response.writeHead(200)
-      response.end(buffer)
+      function writeResponse() {
+        // read file sync so we don't hold open the file creating a race with
+        // the builder (Windows does not allow us to delete while the file is open).
+        buffer = fs.readFileSync(filename)
+        response.writeHead(200)
+        response.end(buffer)
+      }
+
+      if(request.headers.origin) {
+        // Check if origin resolves to a loopback IP, only add CORS headers if that's the case.
+        dns.lookup(url.parse(request.headers.origin).hostname, function(err, address) {
```

Comment:
Do we *need* a DNS lookup here, or do you think we get away with comparing to `localhost`, `127.0.0.1`, `::1` for performance?